WickedEngine/wiScene.cpp

```diff
@@ -2069,67 +2045,17 @@ namespace wi::scene
 	}
 	void Scene::Merge(Scene& other)
 	{
-		names.Merge(other.names);
-		layers.Merge(other.layers);
-		transforms.Merge(other.transforms);
-		hierarchy.Merge(other.hierarchy);
-		materials.Merge(other.materials);
-		meshes.Merge(other.meshes);
-		impostors.Merge(other.impostors);
-		objects.Merge(other.objects);
-		aabb_objects.Merge(other.aabb_objects);
-		rigidbodies.Merge(other.rigidbodies);
-		softbodies.Merge(other.softbodies);
-		armatures.Merge(other.armatures);
-		lights.Merge(other.lights);
-		aabb_lights.Merge(other.aabb_lights);
-		cameras.Merge(other.cameras);
-		probes.Merge(other.probes);
-		aabb_probes.Merge(other.aabb_probes);
-		forces.Merge(other.forces);
-		decals.Merge(other.decals);
-		aabb_decals.Merge(other.aabb_decals);
-		animations.Merge(other.animations);
-		animation_datas.Merge(other.animation_datas);
-		emitters.Merge(other.emitters);
-		hairs.Merge(other.hairs);
-		weathers.Merge(other.weathers);
-		sounds.Merge(other.sounds);
-		inverse_kinematics.Merge(other.inverse_kinematics);
-		springs.Merge(other.springs);
+		for(auto& componentManager_kval : componentLibrary.componentManagers){
```

Comment:
Probably good idea to make the library compatible with `for(auto& componenManager : library) {}`